### DISCUSSION TITLE
Open weui is very slow

### DISCUSSION THREAD

**Discussion by alanqoudif** [open-webui/open-webui#3588] [Q&A]
I've done an investigation on my VPS resources, and I'm not using all of it, about 10 percent of it.
From this we conclude that the slowness comes from the code used to run OpenWeui. Can it be speeded up?
my VPS D
8 vCPU Cores
32GB RAM
32TB bandwidth

**Comment by windkwbs** [comment]
yes,very slow

**Comment by AlgorithmicKing737** [comment]
really need help here @tjbck

**Comment by windkwbs** [comment]
After updating to the latest version, the page still loads slowly and automatic recognition of the browser's default language becomes unstable. @tjbck

**Comment by mirkancal** [comment]
One possible cause could be Nginx buffering, if you're using it. I recently encountered a similar issue where responses were coming in paragraphs instead of token-by-token. Here's how I resolved it:

1. Edit your Nginx config (usually in /etc/nginx/sites-available/):

```nginx
location / {
    proxy_pass http://localhost:YOUR_PORT;
    proxy_buffering off;
    proxy_cache off;
    proxy_set_header Connection '';
    proxy_http_version 1.1;
    chunked_transfer_encoding off;
}
```

2. Test and reload Nginx:
```
sudo nginx -t
sudo systemctl reload nginx
```

This disables buffering and adjusts connection settings, which significantly improved streaming performance for me.

**Comment by tandav** [comment]
Same here. I solved the issue by using postgres (via `DATABASE_URL` environment variable) instead of default SQLite. Now api request to `/chats` (which performed when you open OpenWebUI) is much faster.

**Comment by tjbck** [comment]
https://github.com/open-webui/open-webui/issues/3620
